### PR TITLE
e2e: use origin-pod image in e2e test

### PR DESCRIPTION
**Pull request by mfojtik**
Fixes: https://github.com/openshift/origin/issues/15543

### CODE REVIEW

**Comment by kargakis** [comment]
@mfojtik integration tests flaked, please open an issue or link the existing one
```
failed  TestRootAPIPaths
	panic: duplicate metrics collector registration attempted
```
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin_integration/5312/consoleFull#9938473135898c58db7602c31c0eab717

**Comment by mfojtik** [comment]
@kargakis flake is https://github.com/openshift/origin/issues/15499

File: test/end-to-end/core.sh [hunk 1]
```diff
@@ -566,7 +566,7 @@ os::log::info "Validating image pruning"
 # builder service account should have the power to create new image streams: prune in this case
 os::cmd::expect_success "docker login -u e2e-user -p $(oc sa get-token builder -n cache) -e builder@openshift.com ${DOCKER_REGISTRY}"
 os::cmd::expect_success 'docker pull busybox'
-GCR_PAUSE_IMAGE="gcr.io/google_containers/pause:3.0"
+GCR_PAUSE_IMAGE="openshift/origin-pod:v3.6.0"
```
Comment (stevekuznetsov) [inline test/end-to-end/core.sh]: Does the version matter? If not, don't list it so we use whatever is on the local daemon storage

Comment (mfojtik) [inline test/end-to-end/core.sh]: @stevekuznetsov well :latest might not be always on local storage and not listing it means :latest i guess... this version tag will be there forever I guess?

Comment (stevekuznetsov) [inline test/end-to-end/core.sh]: >  :latest might not be always on local storage

Images are a hard pre-requisite for `hack/test-end-to-end.sh` so we can be certain they will have been built before this test runs. They are built and tagged for the SHA we're testing and `:latest` always. I think it is safe to assume

Comment (smarterclayton) [inline test/end-to-end/core.sh]: This should be `${version}`

Comment (smarterclayton) [inline test/end-to-end/core.sh]: This is a template filled out by the node, where `${version}` comes from the binary.  It has to be double escaped properly here.

Comment (mfojtik) [inline test/end-to-end/core.sh]: fixed.

File: test/end-to-end/core.sh [hunk 2]
```diff
@@ -566,7 +566,7 @@ os::log::info "Validating image pruning"
 # builder service account should have the power to create new image streams: prune in this case
 os::cmd::expect_success "docker login -u e2e-user -p $(oc sa get-token builder -n cache) -e builder@openshift.com ${DOCKER_REGISTRY}"
 os::cmd::expect_success 'docker pull busybox'
-GCR_PAUSE_IMAGE="gcr.io/google_containers/pause:3.0"
+GCR_PAUSE_IMAGE="openshift/origin-pod:\${version}"
```
Comment (stevekuznetsov) [inline test/end-to-end/core.sh]: Should rename the var here too.

**Pull request closed without merging**